cursorless-talon/src/tutorial.py

```diff
@@ -0,0 +1,80 @@
+import json
+import re
+from pathlib import Path
+from typing import Callable
+
+import yaml
+from talon import actions, app
+
+from .get_action_spoken_form import lookup_action
+
+regex = re.compile(r"\{(\w+):([^}]+)\}")
+tutorial_dir = Path(
+    "/Users/pokey/src/cursorless-vscode/src/test/suite/fixtures/recorded/tutorial/unit-2-basic-coding"
+)
+
+
+def process_literal_step(argument: str):
+    return f"<cmd@{argument}/>"
+
+
+def process_action(argument: str):
+    _, spoken_form = lookup_action(argument)
+    return f'<*"{spoken_form}"/>'
+
+
+def process_scope_type(argument: str):
+    _, spoken_form = lookup_scope_type(argument)
+    return f'<*"{spoken_form}"/>'
+
+
+def process_cursorless_command_step(argument: str):
+    step_fixture = yaml.safe_load((tutorial_dir / argument).read_text())
+    return f"<cmd@{cursorless_command_to_spoken_form(step_fixture['command'])}/>"
+
+
+interpolation_processor_map: dict[str, Callable[[str], str]] = {
+    "literalStep": process_literal_step,
+    "action": process_action,
+    "scopeType": process_scope_type,
+    "step": process_cursorless_command_step,
+}
+
+
+def process_tutorial_step(raw: str):
```

Comment:
we should prob arrange for there to just be one call to the extension here. Tbh, though, I wonder if we just let the extension own the tutorial content